pro_tes/middleware/task_distribution.py

```diff
@@ -2,13 +2,22 @@
 
 from copy import deepcopy
 import random
-from typing import List, Optional
+from urllib.parse import urlparse
+from socket import gaierror, gethostbyname
+from typing import Dict, List, Set, Tuple, Optional
+from itertools import combinations
+from ip2geotools.errors import InvalidRequestError
+from ip2geotools.databases.noncommercial import DbIpCity
+from geopy.distance import geodesic
 
 from flask import current_app
 import requests
 
+from pro_tes.middleware.models import AccessUriCombination, TesUriList
 
-def task_distribution() -> Optional[str]:
+
+# pylint: disable-msg=too-many-locals
+def random_task_distribution() -> Optional[str]:
```

Comment:
Both task distribution logic models should have the same signature, but here they return different things (str, List). I think what they should return is a (ranked) list of URLs (not strings).

I am also not sure whether `None` should be acceptable (rather return an empty list)? If there is a specific meaning of `None`, please explain.